### PR TITLE
generic  mapping constrction

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Screenshot 2024-03-28 at 2 55 22 PM

File: src/app/integrations/netsuite/netsuite-main/netsuite-mapping/netsuite-base-mapping/netsuite-base-mapping.component.html [hunk 1]
```diff
@@ -1 +1,12 @@
-<p>netsuite-base-mapping works!</p>
+<app-generic-mapping-v2 *ngIf="!isLoading"
+    [isLoading]="isLoading"
+    [destinationOptions]="destinationOptions"
+    [employeeFieldMapping]="employeeFieldMapping"
+    [sourceField]="sourceField"
+    [destinationField]="destinationField"
+    [showAutoMapEmployee]="showAutoMapEmployee"
+    [appName]="AppName.QBO"
+    [isCategoryMappingGeneric]="sourceField === FyleField.CATEGORY ? true : false"
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-main/netsuite-mapping/netsuite-base-mapping/netsuite-base-mapping.component.html:9]: should be false for netsuite

File: src/app/integrations/netsuite/netsuite-main/netsuite-mapping/netsuite-base-mapping/netsuite-base-mapping.component.html [hunk 2]
```diff
@@ -1 +1,12 @@
-<p>netsuite-base-mapping works!</p>
+<app-generic-mapping-v2 *ngIf="!isLoading"
+    [isLoading]="isLoading"
+    [destinationOptions]="destinationOptions"
+    [employeeFieldMapping]="employeeFieldMapping"
+    [sourceField]="sourceField"
+    [destinationField]="destinationField"
+    [showAutoMapEmployee]="showAutoMapEmployee"
+    [appName]="AppName.QBO"
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-main/netsuite-mapping/netsuite-base-mapping/netsuite-base-mapping.component.html:8]: netsuite

File: src/app/integrations/netsuite/netsuite-main/netsuite-mapping/netsuite-base-mapping/netsuite-base-mapping.component.ts [hunk 3]
```diff
@@ -7,9 +16,88 @@ import { Component, OnInit } from '@angular/core';
...
+      this.isLoading = false;
+      this.toastService.displayToastMessage(ToastSeverity.ERROR, 'Something went wrong, please try again');
+    });
+  }
+
+  private getDestinationField(workspaceGeneralSetting: NetsuiteExportSettingWorkspaceGeneralSettingPost, mappingSettings: MappingSetting[]): string {
+    if (this.sourceField === FyleField.EMPLOYEE) {
+      return workspaceGeneralSetting.employee_field_mapping;
+    } else if (this.sourceField === FyleField.CATEGORY) {
+      return AccountingField.ACCOUNT;
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-main/netsuite-mapping/netsuite-base-mapping/netsuite-base-mapping.component.ts:63]: same rule as dashboard, if er || ccc is ER => expense category, else account

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**